subiquity/server/controllers/ssh.py

```diff
@@ -59,8 +68,52 @@ async def GET(self) -> SSHData:
             install_server=self.model.install_server,
             allow_pw=self.model.pwauth)
 
-    async def POST(self, data: SSHData):
+    async def POST(self, data: SSHData) -> None:
         self.model.install_server = data.install_server
         self.model.authorized_keys = data.authorized_keys
         self.model.pwauth = data.allow_pw
         await self.configured()
+
+    async def fetch_id_GET(self, user_id: str) -> SSHFetchIdResponse:
+        identities: List[SSHIdentity] = []
+
+        import_command = ('ssh-import-id', '--output', '-', '--', user_id)
+        env = None
+        if self.app.base_model.proxy.proxy:
+            env = os.environ.copy()
+            env["https_proxy"] = self.app.base_model.proxy.proxy
+
+        try:
+            cp = await arun_command(import_command, check=True, env=env)
+        except subprocess.CalledProcessError as e:
+            log.exception("ssh-import-id failed. stderr: %s", e.stderr)
+            return SSHFetchIdResponse(status=SSHFetchIdStatus.IMPORT_ERROR,
+                                      identities=None, error=e.stderr)
+        keys_material: str = cp.stdout.replace('\r', '').strip()
+
+        # ssh-keygen supports multiple keys at once.
+        fingerprint_command = ('ssh-keygen', '-l', '-f', '-')
+        try:
+            cp = await arun_command(fingerprint_command, check=True,
+                                    input=keys_material)
+        except subprocess.CalledProcessError as e:
+            log.exception("ssh-import-id failed. stderr: %s", e.stderr)
+            return SSHFetchIdResponse(
+                    tatus=SSHFetchIdStatus.FINGERPRINT_ERROR,
+                    identities=None, error=e.stderr)
+
+        fingerprints: str = cp.stdout.replace(
+                f'#ssh-import-id {user_id}', '').strip()
+
+        zipped = zip(
+                [mat for mat in keys_material.splitlines() if mat],
+                fingerprints.splitlines())
```

Comment:
I'm a tiny bit worried that this might end up matching the key with the wrong fingerprint or (probably worse) end up  dropping a key or a fingerprint if the lists end up mismatched. Maybe it's better to run each key through ssh-keygen individually? (maybe not something for this branch I dunno)